recipes/hwloc/all/conanfile.py

```diff
@@ -0,0 +1,118 @@
+        else:
+            basic_layout(self, src_folder="src")
+
+    def generate(self):
+        if self.settings.os == "Windows":
+            deps = CMakeDeps(self)
+            deps.generate()
+            tc = CMakeToolchain(self)
+            tc.cache_variables["HWLOC_ENABLE_TESTING"] = 'OFF'
+            tc.cache_variables["HWLOC_SKIP_LSTOPO"] = 'ON'
+            tc.cache_variables["HWLOC_SKIP_TOOLS"] = 'ON'
+            tc.cache_variables["HWLOC_SKIP_INCLUDES"] = 'OFF'
+            tc.cache_variables["HWLOC_WITH_OPENCL"] = 'OFF'
+            tc.cache_variables["HWLOC_WITH_CUDA"] = 'OFF'
+            tc.cache_variables["HWLOC_BUILD_SHARED_LIBS"] = self.options.shared
+            tc.cache_variables["HWLOC_WITH_LIBXML2"] = not self.options.disable_libxml2
+            tc.generate()
+        else:
+            deps = AutotoolsDeps(self)
+            deps.generate()
+            tc = AutotoolsToolchain(self)
+            if self.options.disable_libxml2:
+                tc.configure_args.extend(["--disable-libxml2"])
+            tc.configure_args.extend(["--disable-io", "--disable-cairo"])
+            tc.generate()
+
+    def build(self):
+        if self.settings.os == "Windows":
+            cmake = CMake(self)
+            cmake.configure(build_script_folder=os.path.join("contrib", "windows-cmake"))
+            cmake.build()
+        else:
+            autotools = Autotools(self)
+            autotools.configure()
+            autotools.make()
+
+    def package(self):
+        copy(self, "COPYING", self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        if self.settings.os == "Windows":
+            cmake = CMake(self)
+            cmake.install()
+            # remove PDB files
+            rm(self, "*.pdb", os.path.join(self.package_folder, "bin"))
+        else:
+            autotools = Autotools(self)
+            autotools.install()
+            fix_apple_shared_install_name(self)
+            # remove tools
+            rmdir(self, os.path.join(self.package_folder, "bin"))
+
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rm(self, "*.la", os.path.join(self.package_folder, "lib"))
+        rmdir(self, os.path.join(self.package_folder, "share"))
+
+    def package_info(self):
+        self.cpp_info.set_property("pkg_config_name", "hwloc")
+        self.cpp_info.libs = ["hwloc"]
+
+        if not self.options.shared:
+            if self.settings.os in ["Linux", "FreeBSD"]:
+                self.cpp_info.system_libs = ["m"]
+        if is_apple_os(self):
+            self.cpp_info.frameworks = ['IOKit', 'Foundation', 'CoreFoundation']
+
+        # TODO: to remove in conan v2
+        self.cpp_info.filenames["cmake_find_package"] = "hwloc"
+        self.cpp_info.filenames["cmake_find_package_multi"] = "hwloc"
+        self.cpp_info.names["cmake_find_package"] = "hwloc"
+        self.cpp_info.names["cmake_find_package_multi"] = "hwloc"
```

Comment:
```suggestion
```

These can be dropped for new recipes. Only necessary for backward compat of previously existing recipes, but new ones should already use ``CMakeDeps`` to consume this.